### PR TITLE
data provider - cluster names box

### PR DESCRIPTION
**Pull request by pswedrak**

### CODE REVIEW

File: gui/functionality/adjust_timeouts.py [hunk 1]
```diff
@@ -114,6 +117,8 @@ def item_pressed(self, item_pressed):
         self.current_cluster = item_pressed
         self.refresh_timeouts(item_pressed)
     
-    def load_cluster_names(self):
-        load_environment()
-        self.cluster_names = list(EnvironmentProvider().environment.clusters.keys())
+    def handle_cluster_name_change(self):
```
Comment (PiekarzKarolina) [inline gui/functionality/adjust_timeouts.py]: `name_change`? From when we allow to change names of clusters? I suggest something more like `cluster_list_modification` or better, shorter name. @jchyzy any ideas?

**Comment by pswedrak** [comment]
@jchyzy 
1. I think it is better to show clusters in order of addtion (as it is done in idact.conf)
2.  I agree that the name should be changed but notice that the functions are not exactly copied - in different classes we might want (and we do want) to do different things and defining them once only diminishes the flexibility
3. I fixed this
4. Where do you take the value from if you don't catch a signal?

**Comment by jchyzy** [comment]
@pswedrak 

1. I mean that maybe it would be a good idea to put cluster name from arg.json as default value (the one that is displayed when window is loaded) - maybe apart from _remove cluster_. Order can be as you wrote.
2. Now they are the same, please fix that
4. `currentText` and `str` methods (https://stackoverflow.com/questions/6061893/how-do-you-get-the-current-text-contents-of-a-qcombobox)

**Comment by pswedrak** [comment]
@jchyzy
We need to use signals because adding/removing clusters happens asynchronously

**Pull request merged**